azkaban-exec-server/src/main/java/azkaban/execapp/FlowRunnerManager.java

```diff
@@ -906,4 +922,54 @@ private void cleanRecentlyFinished() {
     }
   }
 
+  private class PollingThread extends Thread {
+
+    private final long pollingIntervalMs;
+    private boolean shutdown = false;
+
+    public PollingThread(final long pollingIntervalMs) {
+      this.setName("FlowRunnerManager-Polling-Thread");
+      this.pollingIntervalMs = pollingIntervalMs;
+    }
+
+    @Override
+    public void run() {
+      while (!this.shutdown) {
+        synchronized (this) {
+          try {
+            if (AzkabanExecutorServer.getApp() == null) {
+              continue;
+            }
+
+            final Executor executor = requireNonNull(FlowRunnerManager.this.executorLoader
```

Comment:
technically we won't need executor table any more with change. Are you considering eliminating this dependency?